libs/contrib/Decidable/Order/Strict.idr

```diff
@@ -1,51 +1,57 @@
 ||| An strict preorder (sometimes known as a quasi-order, or an
 ||| ordering) is what you get when you remove the diagonal `{(a,b) | a
 ||| r b , b r a}` from a preorder. For example a < b is an ordering.
-||| This module extends base's Decidable.Order with the strict versions.
+||| This module extends base's Control.Order with the strict versions.
 ||| The interface system seems to struggle a bit with some of the constructions,
 ||| so I hacked them a bit. Sorry.
 module Decidable.Order.Strict
 
-import Decidable.Order
+import Control.Relation
+import Control.Order
 import Decidable.Equality
 
 %default total
 
 public export
-interface StrictPreorder t spo where
-  transitive : (a, b, c : t) -> a `spo` b -> b `spo` c -> a `spo` c
-  irreflexive : (a : t) -> Not (a `spo` a)
+interface Irreflexive ty rel where
+  irreflexive : {x : ty} -> Not (rel x x)
 
 public export
-asymmetric : StrictPreorder t spo => (a, b : t) -> a `spo` b -> Not (b `spo` a)
-asymmetric a b aLTb bLTa = irreflexive a $
-                           Strict.transitive a b a aLTb bLTa
+interface (Transitive ty rel, Irreflexive ty rel) => StrictPreorder  ty rel where
+
+public export
+interface Asymmetric ty rel where
+  asymmetric : {x, y : ty} -> rel x y -> Not (rel y x)
+
+public export
+StrictPreorder ty rel => Asymmetric ty rel where
+  asymmetric xy yx = irreflexive {rel} $ transitive {rel} xy yx
 
 public export
 EqOr : (spo : t -> t -> Type) -> StrictPreorder t spo => (a,b : t) -> Type
 EqOr spo a b = Either (a = b) (a `spo` b)
 
--- Can generalise to an arbitrary equivalence, I belive
-public export
-[MkPreorder] {spo : t -> t -> Type} -> StrictPreorder t spo => Preorder t (EqOr spo) where
-  reflexive a = Left Refl
-  transitive a _ c (Left  Refl) bLTEc        = bLTEc
-  transitive a b _ (Right aLTb) (Left  Refl) = Right aLTb
-  transitive a b c (Right aLTb) (Right bLTc) = Right $ Strict.transitive a b c aLTb bLTc
+-- -- Can generalise to an arbitrary equivalence, I belive
```

Comment:
This file is full of commented out proofs that probably need to be revived.

Reply:
I can't figure out how to get them up to date, and TBH I'm not totally sure what they do. I can cut out the commented stuff or leave it, but as it is now I don't know how to revive them.